core/src/main/java/io/questdb/griffin/engine/join/AbstractJoinRecordCursorFactory.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+package io.questdb.griffin.engine.join;
+
+import io.questdb.cairo.AbstractRecordCursorFactory;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.model.JoinContext;
+
+public abstract class AbstractJoinRecordCursorFactory extends AbstractRecordCursorFactory {
+
+    protected final JoinContext joinContext;
+    protected final RecordCursorFactory masterFactory;
+    protected final RecordCursorFactory slaveFactory;
+
+    public AbstractJoinRecordCursorFactory(RecordMetadata metadata, JoinContext joinContext, RecordCursorFactory masterFactory, RecordCursorFactory slaveFactory) {
+        super(metadata);
+        this.joinContext = joinContext;
+        this.masterFactory = masterFactory;
+        this.slaveFactory = slaveFactory;
+    }
+
+    @Override
+    public RecordCursorFactory getBaseFactory() {
```

Comment:
i think this method could be removed, default in interface returns null too.

Reply:
Removed in latest commits.